### PR TITLE
Life server optimisations

### PR DESCRIPTION
**Pull request by DomT602**
<Please review the guidelines for contributing to this repository. The link is above.>

Resolves #<issue ID here>. <If applicable.>

#### Changes proposed in this pull request: 
- Removing useless stuff (extra parentheses, unneeded variables)
- Private arrays > private commands
- Everything to use params
- _define = if {}
- Used a few 'remoteExecutedOwner' to avoid sending useless info to server when that exists (not done all, as waiting on missionfile to be updated)
- Few other optimisations I can't remember

### CODE REVIEW

File: life_server/Functions/Gangs/fn_updateGang.sqf [hunk 1]
```diff
@@ -5,44 +5,43 @@
...
-        _members = [(_group getVariable "gang_members")] call DB_fnc_mresArray;
-        _owner = _group getVariable ["gang_owner",""];
+        private _bank = [(_group getVariable ["gang_bank",0])] call DB_fnc_numberSafe;
+        private _maxMembers = _group getVariable ["gang_maxMembers",8];
+        private _members = [(_group getVariable "gang_members")] call DB_fnc_mresArray;
+        private _owner = _group getVariable ["gang_owner",""];
         if (_owner isEqualTo "") exitWith {};
 
-        _query = format ["UPDATE gangs SET bank='%1', maxmembers='%2', owner='%3' WHERE id='%4'",_bank,_maxMembers,_owner,_groupID];
+        private _query = format ["UPDATE gangs SET bank='%1', maxmembers='%2', owner='%3' WHERE id='%4'",_bank,_maxMembers,_owner,_groupID];
```
Comment (BoGuu) [inline life_server/Functions/Gangs/fn_updateGang.sqf]: This file is currently relying on _query being defined outside the switch scope (albeit nil). See the last three lines of the file. _query needs to be declared (but not defined) in the parent scope

File: life_server/Functions/Housing/fn_deleteDBContainer.sqf [hunk 2]
```diff
@@ -5,19 +5,20 @@
     Description:
     Delete Container and remove Container in Database
 */
-private ["_house","_houseID","_ownerID","_housePos","_query","_radius","_containers"];
-_container = [_this,0,objNull,[objNull]] call BIS_fnc_param;
+params [
+    ["_container",objNull,[objNull]]
+];
 if (isNull _container) exitWith  {diag_log "container null";};
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_deleteDBContainer.sqf]: Could you remove the diag_log here? It's probably confusing for the end user

File: life_server/Functions/Housing/fn_deleteDBContainer.sqf [hunk 3]
```diff
@@ -5,19 +5,20 @@
...
-    _ownerID = (_container getVariable "container_owner") select 0;
-    _query = format ["UPDATE containers SET owned='0', pos='[]' WHERE pid='%1' AND pos='%2' AND owned='1'",_ownerID,_containerPos];
+    private _containerPos = getPosATL _container;
+    (_container getVariable "container_owner") params ["_ownerID"];
+    private _query = format ["UPDATE containers SET owned='0', pos='[]' WHERE pid='%1' AND pos='%2' AND owned='1'",_ownerID,_containerPos];
     //systemChat format [":SERVER:sellHouse: container_id does not exist"];
 } else {
     //systemChat format [":SERVER:sellHouse: house_id is %1",_houseID];
-    _query = format ["UPDATE containers SET owned='0', pos='[]' WHERE id='%1'",_containerID];
+    private _query = format ["UPDATE containers SET owned='0', pos='[]' WHERE id='%1'",_containerID];
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_deleteDBContainer.sqf]: _query must be declared outside the scope again here, it's used further down

File: life_server/Functions/Housing/fn_fetchPlayerHouses.sqf [hunk 4]
```diff
@@ -8,53 +8,56 @@
...
     //Fix position for more accurate positioning
-    _posX = _position select 0;
-    _posY = _position select 1;
-    _posZ = _position select 2;
-    _currentPos = getPosATL _container;
-    _fixX = (_currentPos select 0) - _posX;
-    _fixY = (_currentPos select 1) - _posY;
-    _fixZ = (_currentPos select 2) - _posZ;
+    _position params ["_posX", "_posY", "_posZ"];
+    getPosATL _container params ["_curPosX", "_curPosY", "_curPosZ"];
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_fetchPlayerHouses.sqf]: Could you add some parenthesis here for clarity?

File: life_server/Functions/Housing/fn_houseCleanup.sqf [hunk 5]
```diff
@@ -5,15 +5,18 @@
     Description:
     Cleans up containers inside in house of player.
 */
-private ["_query","_containers"];
-_query = format ["SELECT pid, pos FROM containers WHERE pid='%1' AND owned='1'",_this];
+private _query = format ["SELECT pid, pos FROM containers WHERE pid='%1' AND owned='1'",_this];
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_houseCleanup.sqf]: I'll mark this to change, as this is quite ambiguous to read without following the execution path (in reference to _this)

File: life_server/Functions/Housing/fn_houseCleanup.sqf [hunk 6]
```diff
@@ -5,15 +5,18 @@
...
-    _pos = call compile format ["%1",_x select 1];
-    _container = nearestObjects[_pos,["Box_IND_Grenades_F","B_supplyCrate_F"],12];
+    _x params [
+        "",
+        "_pos"
+    ];
+    _position = call compile format ["%1",_pos];
+    _container = nearestObjects[_position,["Box_IND_Grenades_F","B_supplyCrate_F"],12];
     {
         deleteVehicle _x;
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_houseCleanup.sqf]: We can move this over to count

File: life_server/Functions/Housing/fn_houseGarage.sqf [hunk 7]
```diff
@@ -12,18 +12,12 @@ params [
...
 
 private _housePos = getPosATL _house;
-private "_query";
 
-if (_mode isEqualTo 0) then {
-    _query = format ["UPDATE houses SET garage='1' WHERE pid='%1' AND pos='%2'",_uid,_housePos];
-} else {
-    _query = format ["UPDATE houses SET garage='0' WHERE pid='%1' AND pos='%2'",_uid,_housePos];
-};
+private _garage = ['0', '1'] select _mode;
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_houseGarage.sqf]: Wrong way around :)

File: life_server/Functions/Housing/fn_initHouses.sqf [hunk 8]
```diff
@@ -27,14 +33,14 @@ for [{_x=0},{_x<=_count},{_x=_x+10}] do {
...
 };
 
-_blacklistedHouses = "count (getArray (_x >> 'garageBlacklists')) > 0" configClasses (missionconfigFile >> "Housing" >> worldName);
-_blacklistedGarages = "count (getArray (_x >> 'garageBlacklists')) > 0" configClasses (missionconfigFile >> "Garages" >> worldName);
+private _blacklistedHouses = "count (getArray (_x >> 'garageBlacklists')) > 0" configClasses (missionconfigFile >> "Housing" >> worldName);
+private _blacklistedGarages = "count (getArray (_x >> 'garageBlacklists')) > 0" configClasses (missionconfigFile >> "Garages" >> worldName);
 _blacklistedHouses = _blacklistedHouses apply {configName _x};
 _blacklistedGarages = _blacklistedGarages apply {configName _x};
 
 for "_i" from 0 to count(_blacklistedHouses)-1 do {
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_initHouses.sqf]: Can make this a count loops, including the forEach below

File: life_server/Functions/Housing/fn_sellHouse.sqf [hunk 9]
```diff
@@ -6,19 +6,20 @@
...
 
-_houseID = _house getVariable ["house_id",-1];
+private _houseID = _house getVariable ["house_id",-1];
 if (_houseID isEqualTo -1) then {
-    _housePos = getPosATL _house;
-    _ownerID = (_house getVariable "house_owner") select 0;
-    _query = format ["UPDATE houses SET owned='0', pos='[]' WHERE pid='%1' AND pos='%2' AND owned='1'",_ownerID,_housePos];
+    private _housePos = getPosATL _house;
+    (_house getVariable "house_owner") params ["_ownerID"];
+    private _query = format ["UPDATE houses SET owned='0', pos='[]' WHERE pid='%1' AND pos='%2' AND owned='1'",_ownerID,_housePos];
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_sellHouse.sqf]: _query is needed out of scope again here

File: life_server/Functions/Housing/fn_sellHouseContainer.sqf [hunk 10]
```diff
@@ -6,17 +6,18 @@
...
 
-_containerID = _container getVariable ["container_id",-1];
+private _containerID = _container getVariable ["container_id",-1];
 if (_containerID isEqualTo -1) then {
-    _containerPos = getPosATL _container;
-    _ownerID = (_container getVariable "container_owner") select 0;
-    _query = format ["UPDATE containers SET owned='0', pos='[]' WHERE pid='%1' AND pos='%2' AND owned='1'",_ownerID,_containerPos];
+    private _containerPos = getPosATL _container;
+    (_container getVariable "container_owner") params ["_ownerID"];
+    private _query = format ["UPDATE containers SET owned='0', pos='[]' WHERE pid='%1' AND pos='%2' AND owned='1'",_ownerID,_containerPos];
```
Comment (BoGuu) [inline life_server/Functions/Housing/fn_sellHouseContainer.sqf]: Ditto

File: life_server/Functions/MySQL/fn_insertRequest.sqf [hunk 11]
```diff
@@ -7,24 +7,21 @@
...
 */
-private ["_queryResult","_query","_alias"];
 params [
-    "_uid",
-    "_name",
+    ["_uid","",[""]],
+    ["_name","",[""]],
     ["_money",-1,[0]],
-    ["_bank",-1,[0]],
-    ["_returnToSender",objNull,[objNull]]
```
Comment (BoGuu) [inline life_server/Functions/MySQL/fn_insertRequest.sqf]: Did you remove the passing of the player object from the client file here?

File: life_server/Functions/MySQL/fn_insertVehicle.sqf [hunk 12]
```diff
@@ -5,20 +5,18 @@
...
     Inserts the vehicle into the database
 */
-private ["_query","_sql"];
 params [
-    "_uid",
-    "_side",
-    "_type",
-    "_className",
+    ["_uid","",[""]],
+    ["_side",civilian,[civilian]],
```
Comment (BoGuu) [inline life_server/Functions/MySQL/fn_insertVehicle.sqf]: This isn't ideal, as we can't change for bad input, sideUnknown perhaps? You're still checking _side == "" below

File: life_server/Functions/MySQL/fn_insertVehicle.sqf [hunk 13]
```diff
@@ -5,20 +5,18 @@
...
+    ["_type","",[""]],
+    ["_className","",[""]],
     ["_color",-1,[0]],
     ["_plate",-1,[0]]
 ];
 
 //Stop bad data being passed.
 if (_uid isEqualTo "" || _side isEqualTo "" || _type isEqualTo "" || _className isEqualTo "" || _color isEqualTo -1 || _plate isEqualTo -1) exitWith {};
```
Comment (BoGuu) [inline life_server/Functions/MySQL/fn_insertVehicle.sqf]: Lazy eval :)

File: life_server/Functions/MySQL/fn_mresArray.sqf [hunk 14]
```diff
@@ -7,18 +7,16 @@
...
 
-for "_i" from 0 to (count _array)-1 do
-{
+for "_i" from 0 to (count _array)-1 do {
     _sel = _array select _i;
-    if (!(_i isEqualTo 0) && !(_i isEqualTo ((count _array)-1))) then
-    {
-        if (_sel isEqualTo 34) then
-        {
+    if (!(_i isEqualTo 0) && !(_i isEqualTo ((count _array)-1))) then {
```
Comment (BoGuu) [inline life_server/Functions/MySQL/fn_mresArray.sqf]: Lazy eval :)

File: life_server/Functions/MySQL/fn_queryRequest.sqf [hunk 15]
```diff
@@ -11,13 +11,10 @@
     ARRAY - If array has 0 elements it should be handled as an error in client-side files.
     STRING - The request had invalid handles or an unknown error and is logged to the RPT.
 */
-private ["_uid","_side","_query","_queryResult","_tickTime","_tmp"];
-_uid = [_this,0,"",[""]] call BIS_fnc_param;
-_side = [_this,1,sideUnknown,[civilian]] call BIS_fnc_param;
-_ownerID = [_this,2,objNull,[objNull]] call BIS_fnc_param;
```
Comment (BoGuu) [inline life_server/Functions/MySQL/fn_queryRequest.sqf]: Ditto as before, is this still being passed?

File: life_server/Functions/MySQL/fn_updateRequest.sqf [hunk 16]
```diff
@@ -6,17 +6,19 @@
...
+    ["_licenses",[],[[]]],
+    ["_gear",[],[[]]],
+    ["_stats",[100,100,0],[[]]],
+    ["_arrested",false,[false]],
+    ["_alive",false,[true]],
+    ["_position",[],[[]]]
+];
 
 //Get to those error checks.
 if ((_uid isEqualTo "") || (_name isEqualTo "")) exitWith {};
```
Comment (BoGuu) [inline life_server/Functions/MySQL/fn_updateRequest.sqf]: lazy eval :)

File: life_server/Functions/Systems/fn_getVehicles.sqf [hunk 17]
```diff
@@ -7,31 +7,32 @@
     Sends a request to query the database information and returns vehicles.
 */
 private ["_pid","_side","_type","_unit","_ret","_tickTime","_queryResult"];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_getVehicles.sqf]: ahhhhhhh

File: life_server/Functions/Systems/fn_getVehicles.sqf [hunk 18]
```diff
@@ -7,31 +7,32 @@
...
+        [[]] remoteExec ["life_fnc_impoundMenu",remoteExecutedOwner];
     };
 };
 
-_unit = owner _unit;
 _side = switch (_side) do {
     case west:{"cop"};
     case civilian: {"civ"};
     case independent: {"med"};
     default {"Error"};
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_getVehicles.sqf]: Perhaps just breakout of the file here, with the below RE

File: life_server/Functions/Systems/fn_vehicleStore.sqf [hunk 19]
```diff
@@ -5,35 +5,34 @@
...
     _damage = [];
 };
 _damage = [_damage] call DB_fnc_mresArray;
 
 // because fuel price!
-if (LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1) then {
-    _fuel = (fuel _vehicle);
-    } else {
-    _fuel = 1;
+private _fuel = if (LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1) then {
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_vehicleStore.sqf]: Select would be neater to use here

File: life_server/Functions/Systems/fn_vehicleStore.sqf [hunk 20]
```diff
@@ -5,35 +5,34 @@
...
+private _resourceItems = LIFE_SETTINGS(getArray,"save_vehicle_items");
 
 if (isNull _vehicle || isNull _unit) exitWith {life_impound_inuse = false; (owner _unit) publicVariableClient "life_impound_inuse";life_garage_store = false;(owner _unit) publicVariableClient "life_garage_store";}; //Bad data passed.
-_vInfo = _vehicle getVariable ["dbInfo",[]];
+private _vInfo = _vehicle getVariable ["dbInfo",[]];
 
 if (count _vInfo > 0) then {
-    _plate = _vInfo select 1;
-    _uid = _vInfo select 0;
+    _vInfo params ["_uid", "_plate"]
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_vehicleStore.sqf]: These need defining outside of the scope

File: life_server/Functions/Systems/fn_vehicleStore.sqf [hunk 21]
```diff
@@ -5,35 +5,34 @@
...
 };
 
 // save damage.
 if (LIFE_SETTINGS(getNumber,"save_vehicle_damage") isEqualTo 1) then {
-    _damage = getAllHitPointsDamage _vehicle;
-    _damage = _damage select 2;
-    } else {
+    getAllHitPointsDamage _vehicle params ["", "", "_damage"];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_vehicleStore.sqf]: Same as above

File: life_server/Functions/Systems/fn_whoDoneIt.sqf [hunk 22]
```diff
@@ -11,19 +11,18 @@ params [
 ];
 
 if (isServer) then {
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_whoDoneIt.sqf]: Can change this to an !exitWith

File: life_server/Functions/Systems/fn_whoDoneIt.sqf [hunk 23]
```diff
@@ -11,19 +11,18 @@ params [
...
-    _killerVeh = vehicle _killer;
-    _distance = _killer distance _victim;
+    private _killerWep = currentWeapon _killer;
+    private _killerVeh = vehicle _killer;
+    private _distance = _killer distance _victim;
     _distance = floor(_distance);
 
-    _message = "";
-    if (_victim == _killer) then {
+    private _message = "";
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_whoDoneIt.sqf]: Can move these over to a `call { if () exitWith{} }` type deal

File: life_server/Functions/MySQL/fn_queryRequest.sqf (hunk 15 above)
Comment (DomT602) [reply life_server/Functions/MySQL/fn_queryRequest.sqf]: Sorted.

File: life_server/Functions/MySQL/fn_insertRequest.sqf (hunk 11 above)
Comment (DomT602) [reply life_server/Functions/MySQL/fn_insertRequest.sqf]: Sorted.

File: life_server/Functions/MySQL/fn_updateRequest.sqf [hunk 24]
```diff
@@ -21,7 +21,7 @@ params [
 ];
 
 //Get to those error checks.
-if ((_uid isEqualTo "") || (_name isEqualTo "")) exitWith {};
+if ({_uid isEqualTo ""} || {_name isEqualTo ""}) exitWith {};
```
Comment (BoGuu) [inline life_server/Functions/MySQL/fn_updateRequest.sqf]: {}

File: life_server/Functions/Systems/fn_vehicleStore.sqf [hunk 25]
```diff
@@ -16,11 +16,13 @@ private _resourceItems = LIFE_SETTINGS(getArray,"save_vehicle_items");
 if (isNull _vehicle || isNull _unit) exitWith {life_impound_inuse = false; (owner _unit) publicVariableClient "life_impound_inuse";life_garage_store = false;(owner _unit) publicVariableClient "life_garage_store";}; //Bad data passed.
 private _vInfo = _vehicle getVariable ["dbInfo",[]];
 
+private ["_uid", "_plate"];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_vehicleStore.sqf]: Define these on separate lines with the keyword

File: life_server/Functions/Systems/fn_whoDoneIt.sqf [hunk 26]
```diff
@@ -10,30 +10,20 @@ params [
     ["_killer",objNull,[objNull]]
 ];
 
-if (isServer) then {
-    if (isNull _victim || isNull _killer) exitWith {};
+if !(isServer) exitWith {};
+if (isNull _victim || isNull _killer) exitWith {};
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_whoDoneIt.sqf]: Lazy eval

**Comment by DomT602** [comment]
Closing this for now due to massive amunt of conflicts which will arise with this PR: https://github.com/AsYetUntitled/Framework/pull/414 
Will work on it again after that PR is confirmed to fully work and merged.

**Pull request closed without merging**